Modules/AccountKit/Sources/Database/DatabaseOperations.swift

```diff
@@ -324,6 +324,57 @@ public class DatabaseOperations {
         .eraseToAnyPublisher()
     }
 
+    public func toggleLibraryItemFavoriteStatus(
+        libraryID: String,
+        subCollection: LibrarySubCollection,
+        itemID: String
+    ) -> AnyPublisher<Void, Error> {
+        let libraryRef = self.database.collection(DatabaseCollection.libraries.rawValue).document(libraryID)
+        let subCollectionRef = libraryRef.collection(subCollection.rawValue)
+
+        return Future<Void, Error> { promise in
+            subCollectionRef
+                .whereField(SavedStory.CodingKeys.id.rawValue, isEqualTo: itemID)
+                .getDocuments { snapshot, error in
+                    if let error {
+                        log.error("Error fetching item with uuid \(itemID) from \(subCollection.rawValue): \(error.localizedDescription)")
+                        promise(.failure(DatabaseError.customError(error.localizedDescription)))
+                        return
+                    }
+
+                    guard let document = snapshot?.documents.first else {
+                        log.error("No item found with uuid \(itemID) in sub-collection \(subCollection.rawValue)")
+                        promise(.failure(DatabaseError.documentNotFound))
+                        return
+                    }
+
+                    let currentIsFavorite = document.data()[SavedStory.CodingKeys.isFavorite.rawValue] as? Bool ?? false
+                    let newFavoriteValue = !currentIsFavorite
```

Comment:
As we discussed, to me, those lines break the Single Responsibility Principle. Because we are in low level file DatabaseOperation, we need the function to be as ignorant as possible and set blindly the value that we give it. This could be done by passing a `isFavorite` boolean argument or by duplicating function and naming them `addLibraryItemToFavorite` and `removeLibraryItemFromFavorite`.